libs/libc/stdio/lib_dtoa_engine.c

```diff
@@ -69,7 +83,7 @@ int __dtoa_engine(double x, FAR struct dtoa_s *dtoa, int max_digits,
   uint8_t flags = 0;
   int i;
 
-  if (__builtin_signbit(x))
+  if (x < 0)
```

Comment:
Can we use `signbit(x)` instead? Ref: https://github.com/apache/nuttx/pull/12811

Reply:
@extinguish please try it.

Reply:
builtins really should not be used in common code because they are compiler specific.  They should be #defined or substituted with logic in include/nuttx/compiler.h to keep the common code compiler independent.  Compiler independence is requires by the "Inviolables"

There are several other locations where builtins are explicitly used in the code.  All of these are non-portable.